### ISSUE TITLE
[FEATURE] Allow MAPINFO "sky2" and "doublesky" commands

### ISSUE THREAD

**Issue by thefatwilly** [odamex/odamex#920]
**Is your feature request related to a problem? Please describe.**
Odamex to allow the mapinfo variables of "sky2" and "doublesky" to give mappers that want to have loads of variety in sky textures to be able to save some space in a wad by utilising the ability to 'graft' together sky textures in the MAPINFO lump. I.E. large map compilations and various CTF projects (like corectf) could seriously benefit from this and make downloading chunky wad files a little less tedious.

**Comment by rakohus** [comment]
@thefatwilly Could you provide some example wads that use these features?

**Comment by thefatwilly** [comment]
This is apparently a rare zdoom feature to be used, so I will have to generate a wad to show. Otherwise most mappers simply use the skybox feature.

**Comment by rakohus** [comment]
![image](https://github.com/odamex/odamex/assets/22926877/271a9644-b0dc-45b7-b921-ecfb25b51ec1)
 Odamex is not currently a huge fan of transparency on the sky, producing a tutti fruitty effect and not rendering the second sky. Afterglow was able to provide me with a test wad. See attached test wad.

[ikdoublesky-test.zip](https://github.com/user-attachments/files/16047521/ikdoublesky-test.zip)

**Comment by rakohus** [comment]
https://github.com/odamex/odamex/assets/22926877/8ec091c1-f6ed-4081-9bd1-8fea163aabdf

This is how it looks in GZ on the software renderer

**Comment by rakohus** [comment]
Possibly relevant here:

GooberMan: Hexen skies don't use normal column based transparency, they treat palette index 0 as transparent
GooberMan: That'd be where the tutti fruitti comes from

**Comment by bcahue** [comment]
I have this almost solved by seeing how it was done in the ZDoom 1.22 source, the last bit is, as GooberMan mentioned, is making a composite sky by comparing the posts in a column and only allowing sky2 to show thru if the sky1 color isn't 0 (cyan).

Here's what I have so far, but it makes the sky all 1 color before overflowing the buffer with skypost because of my laughable understanding of making this work with tallposts:

```c++
// create composite of both skies
byte composite[MAXWIDTH]; // column replacement
byte* skypost;
byte* skypost2;
byte* dest;
int count;
int top;
int bottom;

top = dcol.texturefrac >> FRACBITS;
bottom = (dcol.texturemid + (dcol.yh - centery) * dcol.iscale) >> FRACBITS;
count = bottom - top + 1;

skypost = R_GetTextureColumnData(frontskytex, colnum) + top;
colnum = ((((viewangle + xtoviewangle[x])^skyflip)>>sky2shift) + back_offset)>>16;
skypost2 = R_GetTextureColumnData(backskytex, colnum) + top;
dest = composite + top;

do
{
	if (*skypost)
	{
		*dest++ = *skypost++;
		skypost2++;
	}
	else
	{
		*dest++ = *skypost2++;
		skypost++;
	}
}
while (--count);

skyposts[x] = (tallpost_t*)composite + (bottom - top + 1);
```

**Comment by bcahue** [comment]
This is completed:

![image](https://github.com/odamex/odamex/assets/2531014/6a31d4aa-7429-4553-8eb5-761bd8ce35aa)
